actionpack/test/dispatch/request_test.rb

```diff
@@ -23,14 +19,23 @@ def url_for(options = {})
   private
     def stub_request(env = {})
       ip_spoofing_check = env.key?(:ip_spoofing_check) ? env.delete(:ip_spoofing_check) : true
+      ActionDispatch::Http::URL.tld_length = env.delete(:tld_length) if env.key?(:tld_length)
```

Comment:
Moved this line earlier since using Rack::MockRequest also means that any symbol keys being passed to setup the env, will be discarded. [Only string keys are copied.](https://github.com/rack/rack/blob/444dc8a130663159ed490f25948b73ff9b5fbba0/lib/rack/mock_request.rb#L156)